sycl/doc/extensions/proposed/sycl_ext_oneapi_group_load_store_extended.asciidoc

```diff
@@ -0,0 +1,413 @@
+size required for each sub-group. If other scope values are passed, behavior is
+unspecified.
+
+== Data placement
+
+anchor:data_placement[]
+
+To specify a correct data placement for placing of resulting data
+there is a enum (proposed in link:https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_group_sort.asciidoc[`group_sort extention`]):
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+// Properties:
+enum class group_algorithm_data_placement {
+  blocked,
+  striped
+};
+
+}
+----
+
+Data placement layout example on group_load:
+
+* ElementsPerWorkItem = 4
+* 3 work-items in the group
+* input is: in_ptr[] = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11}
+
+Consider 2 layouts:
+
+1.`sycl::ext::oneapi::experimental::group_algorithm_data_placement::blocked`.
+
+|===
+|Work-item id|Output stored in private fixed-size array
+
+|0
+|{0, 1, 2, 3}
+|1
+|{4, 5, 6, 7}
+|2
+|{8, 9, 10, 11}
+|===
+
+2.`sycl::ext::oneapi::experimental::group_algorithm_data_placement::striped`.
+
+|===
+|Work-item id|Output stored in private fixed-size array
+
+|0
+|{0, 3, 6, 9}
+|1
+|{1, 4, 7, 10}
+|2
+|{2, 5, 8, 11}
+|===
+
+There is a property that satisfies
+link:sycl_ext_oneapi_properties.asciidoc[SYCL Properties Extension]
+requirements:
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental::property
+{
+    template<group_algorithm_data_placement type>
+    struct data_placement; // (9)
+}
+----
+
+(9) Specifies data layout used in group_load/store for `sycl::vec` or fixed-size
+arrays functions.
+
+Example:
+`group_load(g, input, output_span, properties<data_placement<blocked>>{});`
+
+== Usage Examples
+
+1.Example shows the simplest case without local memory usage of blocked load
+of global memory from `input` to the private array `data` and store it back to
+`output`
+
+[source,c++]
+----
+namespace sycl_exp = sycl::ext::oneapi::experimental;
+
+q.submit([&](sycl::handler& cgh) {
+    cgh.parallel_for(
+        sycl::nd_range<1>(global_size, local_size),
+        [=](sycl::nd_item<1> item) {
+            T data[items_per_thread];
+
+            sycl_exp::group_load(item.get_group(), input, sycl::span{ data });
+
+            // Work with data...
+
+            sycl_exp::group_store(item.get_group(), output, sycl::span{ data });
+        });
+});
+----
+
+2.Example shows the simple case of blocked load of global memory from `input` to
+the private array `data` and store it back to `output`
+The temporary memory is allocated via `sycl::local_accessor`
+
+[source,c++]
+----
+namespace sycl_exp = sycl::ext::oneapi::experimental;
+
+q.submit([&](sycl::handler& cgh) {
+    constexpr auto temp_memory_size = sycl_exp::memory_required<T, items_per_thread>(
+        sycl::memory_scope::work_group, block_size);
+    sycl::local_accessor<std::byte> buf(temp_memory_size, cgh);
+    cgh.parallel_for(
+        sycl::nd_range<1>(global_size, local_size),
```

Comment:
Should there be some relation between `block_size` and `local_size`?

Reply:
fixed it in this example, should be same for work-group case, for sub-group case I assume it should be something like local_size = block_size * number of subgroups, but I am not sure performance-wise